lib/omnibus/health_check.rb

```diff
@@ -459,20 +459,75 @@ def health_check_otool
       current_library = nil
       bad_libs = {}
 
-      read_shared_libs("find #{project.install_dir}/ -type f | egrep '\.(dylib|bundle)$' | xargs otool -L") do |line|
+      yield_shellout_results("find #{project.install_dir}/ -type f | egrep '\.(dylib|bundle)$' | xargs otool -L") do |line|
         case line
         when /^(.+):$/
           current_library = Regexp.last_match[1]
         when /^\s+(.+) \(.+\)$/
           linked = Regexp.last_match[1]
           name = File.basename(linked)
-          bad_libs = check_for_bad_library(bad_libs, current_library, name, linked)
+          bad_libs = check_for_bad_macos_library(bad_libs, current_library, name, linked)
         end
       end
 
       bad_libs
     end
 
+    #
+    # Check the given path and library for "bad" libraries.
+    #
+    def check_for_bad_macos_library(bad_libs, current_library, name, linked)
+      safe = nil
+
+      whitelist_libs = MAC_WHITELIST_LIBS
+
+      whitelist_libs.each do |reg|
+        safe ||= true if reg.match(name)
+      end
+
+      whitelist_files.each do |reg|
+        safe ||= true if reg.match(current_library)
+      end
+
+      log.debug(log_key) { "  --> Dependency: #{name}" }
+      log.debug(log_key) { "  --> Provided by: #{linked}" }
+
+      linked_present = false
+
+      if !safe
+        rpath_regexp = Regexp.new("@rpath")
+        install_dir_regexp = Regexp.new(project.install_dir)
+
+        # Do the linker's work of replacing @rpath with the rpaths defined by the library
+        if linked =~ rpath_regexp
+          possible_paths = []
+          yield_shellout_results("otool -l #{current_library} | grep LC_RPATH -A2 | grep path | awk '{ print $2 }'") do |rpath|
```

Comment:
I'm going to trust you on this :)

Reply:
If you have a minute, it would be quite nice if you could add a small comment with some sample output for the:
```otool -l #{current_library}```
command, that'd provide nice context to make the rest of the function easier to follow.

Reply:
Yeah I can try, the only thing is the output of this command takes more than a hundred lines with many different things inside (mach headers, load commands, special sections, ...) 😅.
I can add to the code a sample the relevant part (the load commands), which look like this (from `libdatadog-agent-two.dylib`):
```
Load command 13
          cmd LC_LOAD_DYLIB
      cmdsize 56
         name /usr/lib/libSystem.B.dylib (offset 24)
   time stamp 2 Thu Jan  1 01:00:02 1970
      current version 1238.60.2
compatibility version 1.0.0
Load command 14
          cmd LC_RPATH
      cmdsize 48
         name /opt/datadog-agent/embedded/lib (offset 24)
```